src/components/balances/CurrencySelect/__tests__/useCurrencies.test.ts

```diff
@@ -0,0 +1,56 @@
+import { act, renderHook } from '@testing-library/react'
+import { getFiatCurrencies } from '@safe-global/safe-gateway-typescript-sdk'
+import useCurrencies from '../useCurrencies'
+import { Errors, logError } from '@/services/exceptions'
+
+jest.mock('@safe-global/safe-gateway-typescript-sdk', () => ({
+  getFiatCurrencies: jest.fn(),
+}))
+
+jest.mock('@/services/exceptions', () => {
+  const originalModule = jest.requireActual('@/services/exceptions')
+
+  return {
+    ...originalModule, // this will keep the original exports
+    logError: jest.fn(), // this will override logError with a mock function
+  }
+})
+
+describe('useCurrencies', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('should fetch and return the list of ISO 4217 currencies', async () => {
+    const iso4217Currencies = ['USD', 'EUR', 'GBP']
+    const mockCurrencies = [...iso4217Currencies, 'SATS']
+    const getFiatCurrenciesMock = jest.fn(async () => Promise.resolve(mockCurrencies))
+    const myMock = getFiatCurrencies as jest.Mock
+    myMock.mockImplementation(getFiatCurrenciesMock)
+
+    const { result, rerender } = renderHook(() => useCurrencies())
+
+    expect(result.current).toBeUndefined()
+
+    await act(async () => {
+      await rerender()
+    })
+
+    expect(result.current).toEqual(iso4217Currencies)
+    expect(getFiatCurrenciesMock).toHaveBeenCalledTimes(1)
```

Comment:
nit: I am getting an info that await is not necessary here. I think it "promisifies" the expression which makes it work?
<img width="536" alt="Screenshot 2024-01-05 at 11 51 30" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/db8579af-c979-434b-94bb-2ea297a455ca">

An alternative is to use `await waitFor` on the expect expression e.g.

```
act(() => {
  rerender()
}

await waitFor(() => {
  expect(...)
}
```

Reply:
with
```
act(() => {
  rerender()
}
```
the test fails

```
async act(() => {
  rerender()
}
```
works, but the IDEA complains as well:
<img width="566" alt="grafik" src="https://github.com/safe-global/safe-wallet-web/assets/693770/c6de8e01-0ed8-42ed-bfe7-af42ed5c467f">

and then:
```
    await act(async () => {
      rerender()
    })
```

this works and the IDE doesn't complain.

Reply:
It even works entirely without `act` as long as we wrap the `expect` call(s) in an `await waitFor`.